prowler/providers/azure/services/keyvault/keyvault_service.py

```diff
@@ -0,0 +1,148 @@
+from azure.mgmt.keyvault import KeyVaultManagementClient
+from azure.mgmt.keyvault.v2023_07_01.models import (
+    KeyAttributes,
+    SecretAttributes,
+    VaultProperties,
+)
+
+from prowler.lib.logger import logger
+from prowler.providers.azure.lib.service.service import AzureService
+
+
+########################## Storage
+class KeyVault(AzureService):
+    def __init__(self, audit_info):
+        super().__init__(KeyVaultManagementClient, audit_info)
+        self.key_vaults = self.__get_key_vaults__(audit_info)
+
+    def __get_key_vaults__(self, audit_info):
+        logger.info("KeyVault - Getting key_vaults...")
+        key_vaults = {}
+        for subscription, client in self.clients.items():
+            try:
+                key_vaults.update({subscription: []})
+                key_vaults_list = client.vaults.list()
+                for keyvault in key_vaults_list:
+                    resource_group = keyvault.id.split("/")[4]
+                    keyvault_name = keyvault.name
+                    keyvault_properties = client.vaults.get(
+                        resource_group, keyvault_name
+                    ).properties
+                    keys = self.__get_keys__(
+                        subscription, resource_group, keyvault_name, audit_info
+                    )
+                    secrets = self.__get_secrets__(
+                        subscription, resource_group, keyvault_name
+                    )
+                    key_vaults[subscription].append(
+                        KeyVaultInfo(
+                            id=getattr(keyvault, "id", None),
+                            name=getattr(keyvault, "name", None),
+                            location=getattr(keyvault, "location", None),
+                            resource_group=resource_group,
+                            properties=keyvault_properties,
+                            keys=keys,
+                            secrets=secrets,
+                        )
+                    )
+            except Exception as error:
+                logger.error(
+                    f"Subscription name: {subscription} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+                )
+        return key_vaults
+
+    def __get_keys__(self, subscription, resource_group, keyvault_name, audit_info):
+        logger.info(f"KeyVault - Getting keys for {keyvault_name}...")
+        keys = []
+        try:
+            client = self.clients[subscription]
+            keys_list = client.keys.list(resource_group, keyvault_name)
+            for key in keys_list:
+                keys.append(
+                    Key(
+                        id=getattr(key, "id", None),
+                        name=getattr(key, "name", None),
+                        enabled=getattr(key.attributes, "enabled", None),
+                        location=getattr(key, "location", None),
+                        attributes=getattr(key, "attributes", None),
```

Comment:
Are we handling the cases when these attributes are None?

Reply:
Done! Good catch